scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -1024,17 +1097,85 @@ public boolean killJob(JobId jobId) throws NotConnectedException, UnknownJobExce
     @Override
     public void changeJobPriority(JobId jobId, JobPriority priority)
             throws NotConnectedException, UnknownJobException, PermissionException, JobAlreadyFinishedException {
+        blockIfJobIsStillUnderSubmission(jobId);
+
         frontendState.checkChangeJobPriority(jobId, priority);
 
         schedulingService.changeJobPriority(jobId, priority);
     }
 
+    /**
+     * This method blocks until job is actually finished its submission.
+     *
+     * @param jobId
+     */
+    private void blockIfJobIsStillUnderSubmission(JobId jobId) throws UnknownJobException {
+        synchronized (waitForJobsToBeSubmitted) {
+            while (waitForJobsToBeSubmitted.contains(jobId)) {
+                try {
+                    waitForJobsToBeSubmitted.wait();
+                } catch (InterruptedException e) {
+                    String message = "Something tried to interrupt our blocking";
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)